src/app/clarity/reducer.ts

```diff
@@ -0,0 +1,40 @@
+import { Reducer } from 'redux';
+import { ActionType, getType } from 'typesafe-actions';
+import * as actions from './actions';
+import { ClarityDescription } from './descriptionInterface';
+
+export type ClarityAction = ActionType<typeof actions>;
+
+export interface ClarityState {
+  /**
+   * Descriptions from cumunity provided by Clarity API
+   */
+  descriptions?: ClarityDescription;
+  // I will add more things i need for clarity later on
+}
+
+export const fetchClarityDescriptions_ = async () => {
+  try {
+    const data = await fetch('https://ice-mourne.github.io/database-clarity/descriptions.json');
+    const json: ClarityDescription = await data.json();
+    return json;
```

Comment:
We can probably cache this in IndexedDB and refresh it on a TTL - maybe 24 hours? I can do that as a followup

Reply:
I have plans on making separate file for versions you could check before updating because we often update multiple times a day after updates to d2